File: higher-order-binary-ops/less-than-100.go
package main

import "fmt"

func lt(ro int) func(int) bool {
  return func(lo int) bool { return lo < ro }
}

func main() {
  num :=  []int{5, 500, 50, 101, 200, 35}
  var out []int

  for i, _ := range num {
    if (lt(100)(num[i])) {
      out = append(out, num[i])
    }
  }

  fmt.Println("less than 100:", out)
  // => less than 100: [5 50 35]
}

File: higher-order-binary-ops/less_than_100.erl
-module(less_than_100).
-import(lists, [filter/2]).
-export([start/0]).

start() ->
  Lt  = fun(Ro) -> fun(Lo) -> Lo < Ro end end,
  Num = [5, 500, 50, 101, 200, 35],
  Out = lists:filter(Lt(100), Num),
  io:format("less than 100: ~p~n", [Out]).
  %=> less than 100: [5,50,35]
